src/js/libgeo/GeoOps.js

```diff
@@ -625,6 +625,144 @@ geoOps.PointOnSegment.updatePosition = function(el) {
+//    console.log("pB",niceprint(prodB));
+    if (crB[0].value.real < 0) prodB = CSNumber.neg(prodB);
+
+    
+    var erg;
+    if(prodA.value.real > 0 && prodB.value.real > 0){
+        console.log("hit1");
+        var d1 = List.projectiveDistMinScal(P, isec[0]);
+        var d2 = List.projectiveDistMinScal(P, isec[1]);
+
+        erg = d1 < d2 ? isec[0] : isec[1];
+    }
+    else if (prodA.value.real < 0 && prodB.value.real < 0){
+        console.log("hit rand proj");
+    var d1 = List.projectiveDistMinScal(pos, arc.startPoint);
+    var d2 = List.projectiveDistMinScal(pos, arc.endPoint);
+
+    erg = d1 < d2 ? arc.startPoint : arc.endPoint;
+    }
+    else if(prodA.value.real < 0){ 
+        console.log("hit3");
+        erg = isec[1];
+    }
+    else //(prodB.value.real < 0)
+    {
+        console.log("hit4");
+        erg = isec[0];
+    }
+
+
+    return List.normalizeMax(erg);
+};
+
+
+geoOps.PointOnArc = {};
+geoOps.PointOnArc.kind = "P";
+geoOps.PointOnArc.signature = ["C"];
+geoOps.PointOnArc.isMovable = true;
+geoOps.PointOnArc.initialize = function(el) {
+    var arc = csgeo.csnames[el.args[0]];
+    var p = geoOps._helper.initializePoint(el);
+    p = geoOps._helper.projectPointToArc(arc, p);
+
+    var cr = geoOps._helper.PointOnArcCr(arc, p);
+    cr = CSNumber.mult(cr[0],cr[1]);
+    putStateComplexNumber(cr);
+};
+geoOps.PointOnArc.getParamForInput = function(el, pos) {
+    var arc = csgeo.csnames[el.args[0]];
+    var cr = geoOps._helper.PointOnArcCr(arc, pos);
+    cr = CSNumber.mult(cr[0],cr[1]);
+
+    return cr;
+};
+geoOps.PointOnArc.getParamFromState = function(el) {
+    return getStateComplexNumber();
+};
+geoOps.PointOnArc.putParamToState = function(el, param) {
+    putStateComplexNumber(param);
+};
+geoOps.PointOnArc.updatePosition = function(el,isMover) {
+    var arc = csgeo.csnames[el.args[0]];
+    var A = arc.startPoint;
+    var C = arc.endPoint;
+    var B = arc.viaPoint;
+    var II = List.ii;
+
+    var conic = arc.matrix;
+
+    var cr = getStateComplexNumber();
+    putStateComplexNumber(cr); // copy parameter
+
+    var erg;
+    if( isMover){
+        var point = geoOps._helper.initializePoint(el);
+        console.log(niceprint(List.normalizeZ(point)));
+        erg = geoOps._helper.projectPointToArc(arc,point);
+        // projection should be done here
+    }
+    else{
+
+    // l = ([A,C,I]*B-cr*[A,B,I]*C) x I
+    var aciB = List.det(List.turnIntoCSList([A,C,II]))
+        aciB = List.scalmult(aciB,B);
+
+    var dabiC = List.det(List.turnIntoCSList([A,B,II]));
+        dabiC = CSNumber.mult(CSNumber.neg(cr),dabiC);
+        dabiC = List.scalmult(dabiC, C);
+
+    var ll = List.sub(aciB,dabiC);
+        ll = List.normalizeMax(ll);
+        ll = List.cross(ll,List.ii);
+
+    erg = geoOps._helper.IntersectLC(ll, arc.matrix);
+    var d1 = List.projectiveDistMinScal(erg[0], II);
+    var d2 = List.projectiveDistMinScal(erg[1], II);
+
+    erg = d1 < d2 ? erg[1] : erg[0];
```

Comment:
I don't really understand `aciB`, `dabiC` and `ll`, and I fail to understand the comment at first glance. But I dislike having to do a generic line-conic intersection here.

This is how I see it: point `II` sees `A`, `B`, `C` and our desired point `P` under a given cross ratio. So compute lines joining `II` to `A`, `B` and `C`. Scale them such that two representatives form a projective basis. Express the line from `II` to `P` with respect to this basis. So now you have a line and one point on that line (namely `II`), so you want the other point. Which is a *linear*  problem! Ah, I see you compare results against `II`, so I guess you do something similar there but without making use of the one known solution. I see that [`OtherIntersection`](https://github.com/CindyJS/CindyJS/blob/v0.7.3/src/js/libgeo/GeoOps.js#L1612-L1620) does something similar as well. We really should implement a nice function to make use of the essentially linear nature of this problem.